android/demos/compose/src/main/kotlin/com/uber/rib/compose/util/EventStream.kt

```diff
@@ -15,13 +15,18 @@
  */
 package com.uber.rib.compose.util
 
-import com.jakewharton.rxrelay2.PublishRelay
-import io.reactivex.Observable
+import com.uber.rib.core.RibDispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.withContext
 
 class EventStream<T> {
-  private val eventRelay = PublishRelay.create<T>()
+  private val sharedFlow = MutableSharedFlow<T>()
 
-  fun notify(event: T) = eventRelay.accept(event)
+  suspend fun notify(event: T) = withContext(RibDispatchers.Main) {
```

Comment:
how come this uses Main but StateStream's dispatch method doesn't. it'd be really nice to have method calls to notify not have to launch with a coroutinescope similar to StateStream.dispatch()

Reply:
It doesn't and will be removed so much StateStream.

 However, unless we construct with eventStreams with a coroutineScope and they launch them internally (not recommended since we pass these outside of the parent's lifecycle) then they need to be suspending, which means they need to be accessed from a caller using a coroutine. Remember that coroutines are cheap, so it's not a problem like if we had to do it with threads.